detox/src/devices/DeviceRegistry.js

```diff
@@ -64,6 +75,11 @@ class DeviceRegistry {
     return result;
   }
 
+  readRegisteredDevicesUNSAFE() {
+    const contents = fs.readFileSync(this._lockfilePath, readOptions);
+    return JSON.parse(contents);
+  }
```

Comment:
Note: This is effectively needed - not because the unsafe access is mandatory here, but really for making the _sync_ read. From within a signal-catch callback, no further queued work is honored, and hence promises (and thus async-await) cannot be used.